barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_verifier.cpp

```diff
@@ -2,58 +2,148 @@
+    }
+    index++;
+
+    for (auto it = instances.begin() + 1; it != instances.end(); it++, index++) {
         auto inst = *it;
         auto domain_separator = std::to_string(index);
-        inst->instance_size = transcript->template receive_from_prover<uint32_t>(domain_separator + "_circuit_size");
-        inst->public_input_size =
-            transcript->template receive_from_prover<uint32_t>(domain_separator + "_public_input_size");
-        inst->pub_inputs_offset =
-            transcript->template receive_from_prover<uint32_t>(domain_separator + "_pub_inputs_offset");
-
-        for (size_t i = 0; i < inst->public_input_size; ++i) {
-            auto public_input_i =
-                transcript->template receive_from_prover<FF>(domain_separator + "_public_input_" + std::to_string(i));
-            inst->public_inputs.emplace_back(public_input_i);
-        }
-        auto [eta, beta, gamma] = challenges_to_field_elements<FF>(transcript->get_challenges(
-            domain_separator + "_eta", domain_separator + "_beta", domain_separator + "_gamma"));
-
-        const FF public_input_delta = compute_public_input_delta<Flavor>(
-            inst->public_inputs, beta, gamma, inst->instance_size, inst->pub_inputs_offset);
-        const FF lookup_grand_product_delta = compute_lookup_grand_product_delta<FF>(beta, gamma, inst->instance_size);
-        inst->relation_parameters =
-            RelationParameters<FF>{ eta, beta, gamma, public_input_delta, lookup_grand_product_delta };
-        inst->alpha = transcript->get_challenge(domain_separator + "_alpha");
+        receive_and_finalise_instance(inst, domain_separator);
     }
 }
 
 template <class VerifierInstances>
-VerifierFoldingResult<typename VerifierInstances::Flavor> ProtoGalaxyVerifier_<
-    VerifierInstances>::fold_public_parameters(std::vector<uint8_t> fold_data)
+bool ProtoGalaxyVerifier_<VerifierInstances>::verify_folding_proof(std::vector<uint8_t> fold_data)
 {
-    using Flavor = typename VerifierInstances::Flavor;
-
     prepare_for_folding(fold_data);
-    FF delta = transcript->get_challenge("delta");
+
+    auto delta = transcript->get_challenge("delta");
     auto accumulator = get_accumulator();
-    auto log_instance_size = static_cast<size_t>(numeric::get_msb(accumulator->instance_size));
-    auto deltas = compute_round_challenge_pows(log_instance_size, delta);
-    std::vector<FF> perturbator_coeffs(log_instance_size + 1);
-    for (size_t idx = 0; idx <= log_instance_size; idx++) {
+    auto deltas = compute_round_challenge_pows(accumulator->log_instance_size, delta);
+
+    std::vector<FF> perturbator_coeffs(accumulator->log_instance_size + 1);
+    for (size_t idx = 0; idx <= accumulator->log_instance_size; idx++) {
         perturbator_coeffs[idx] = transcript->template receive_from_prover<FF>("perturbator_" + std::to_string(idx));
     }
+    assert(perturbator_coeffs[0] == accumulator->folding_parameters.target_sum);
```

Comment:
Will you use our `ASSERT` instead?

Reply:
didn't know there's a difference, thanks